modules/core/02-client/types/router.go

```diff
@@ -0,0 +1,59 @@
+package types
+
+import (
+	"fmt"
+
+	storetypes "cosmossdk.io/store/types"
+
+	"github.com/cosmos/ibc-go/v8/modules/core/exported"
+)
+
+// Router is a map from a clientType to a LightClientModule instance.
+// The router has a reference to the client store provider (02-client keeper)
+// and will register the store provider on a client module upon route registration.
+type Router struct {
+	routes        map[string]exported.LightClientModule
+	storeProvider exported.ClientStoreProvider
+}
+
+// NewRouter returns an instance of the Router.
+func NewRouter(key storetypes.StoreKey) *Router {
+	return &Router{
+		routes:        make(map[string]exported.LightClientModule),
+		storeProvider: NewStoreProvider(key),
+	}
+}
+
+// AddRoute adds LightClientModule for a given module name. It returns the Router
+// so AddRoute calls can be linked. It will panic if the Router is sealed.
+// The store provider will be registered on the light client module.
+func (rtr *Router) AddRoute(clientType string, module exported.LightClientModule) *Router {
```

Comment:
note: we don't do validation of the clientType string

Reply:
I added validation in https://github.com/cosmos/ibc-go/pull/5806/commits/16e141bacc1001003546f328c81b0bd1d113937b.